pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -130,21 +137,20 @@ library BasePoolMath {
             newBalances[i] = currentBalances[i] + exactAmounts[i];
         }
 
-        // Calculate the invariant using the current balances (before the addition).
-        uint256 currentInvariant = computeInvariant(currentBalances);
-
-        // Calculate the new invariant using the new balances (after the addition).
-        uint256 newInvariant = computeInvariant(newBalances);
-
         // Calculate the new invariant ratio by dividing the new invariant by the old invariant.
-        uint256 invariantRatio = newInvariant.divDown(currentInvariant);
+        uint256 currentInvariant = pool.computeInvariant(currentBalances);
+        // Round down to make `taxableAmount` larger below.
+        uint256 invariantRatio = pool.computeInvariant(newBalances).divDown(currentInvariant);
```

Comment:
we can save `pool.computeInvariant(newBalances)` in var to use it below

Reply:
the problem is that `newBalances` are supposed to change between this line and when the invariant is recalculated, so we can't reuse it.